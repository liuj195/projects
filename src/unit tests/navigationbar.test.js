import React from "react";
import NavigationBar from "../components/navbar/navigationbar";
import ShallowRenderer from "react-test-renderer/shallow";
import { configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

configure({ adapter: new Adapter() });

test("Navbar renders correctly", () => {
  const renderer = new ShallowRenderer();
  const component = renderer.render(
    <NavigationBar
      currentGraph={{}}
      onUpload={() => {}}
      navBarControl={[{ name: "File", disabled: false }]}
      user={{ userID: "abc", access: [] }}
      onLegendClick={() => {}}
    />
  );
  expect(component).toMatchSnapshot();
});
