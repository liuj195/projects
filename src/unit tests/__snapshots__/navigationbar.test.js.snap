// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar renders correctly 1`] = `
<header
  className="usa-header usa-header-basic"
  role="banner"
>
  <div
    className="usa-nav-container"
  >
    <nav
      className="usa-nav"
      role="navigation"
    >
      <ul
        className="usa-nav-primary usa-accordion"
      >
        <DropdownItem
          isDisabled={false}
          itemList={
            Array [
              "Save Graph",
              "Load Graph",
              "Print Graph",
            ]
          }
          name="File"
          onChangeLayout={[Function]}
          onLegendClick={[Function]}
          onSaveFile={[Function]}
          onUpload={[Function]}
          printFile={[Function]}
          styleProps={
            Object {
              "color": "white",
            }
          }
          subMenus={
            Object {
              "Layout": Array [
                "Hierarchic(Dagre)",
                "Circular",
                "Cola",
                "Grid",
              ],
              "Show_Hide": Array [
                "Names",
                "TIN",
                "Tax Period",
              ],
            }
          }
          user={
            Object {
              "access": Array [],
              "userID": "abc",
            }
          }
        />
      </ul>
    </nav>
  </div>
</header>
`;
